main.c

```diff
@@ -1,3 +1,48 @@
+#include <stdint.h>
+
+#include "am1808.h"
+
+// Shared RAM (control input)
+typedef struct shared_ram {
+    // Declare this as a u32 to force more-efficient codegen
+    uint32_t pwms;
+} shared_ram;
+// XXX: The real address in use here is 0x80010000
+// There appears to be a compiler bug where ctable entries with the MSB set
```

Comment:
Do we have a link for the bug we could add here?

Reply:
I'm working on filing a bug right this moment, but unfortunately GCC's bug tracker seems to want manual approval for new account creation.

I have produced a minimal testcase though:

```c
#pragma ctable_entry 30 0x00beef00
void good() {
    *(volatile unsigned int *)0x00beef00 = 0xdead;
}

#pragma ctable_entry 31 0x80beef00
void bad() {
    *(volatile unsigned int *)0x80beef00 = 0xf00d;
}
```

Reply:
I just filed https://gcc.gnu.org/bugzilla/show_bug.cgi?id=121124 to track this issue.